### DISCUSSION TITLE
Expected behaviour? internal_metrics values monotonically increasing even after taking monotonic difference

### DISCUSSION THREAD

**Discussion by leongshengmin** [vectordotdev/vector#14730] [Q&A]
```
monotonic_diff(sum:vector.component_sent_event_bytes_total{component_kind:source,component_id:internal_metrics} by {component_id}.as_count())
```
Currently using this datadog query but observing the values monotonically increasing even when using the `monotonic_diff` function (which graphs the delta of the metric like diff() but only if the delta is positive). We are not sure if this is expected behavior or otherwise as we are only noticing this for `internal_metrics` source but not `kubernetes_logs` source?

<img width="1553" alt="Screenshot 2022-10-05 at 6 00 28 PM" src="https://user-images.githubusercontent.com/35290582/194035691-53894e2a-36ce-46bb-b111-de12688ec8ac.png">

<br class="Apple-interchange-newline">

**Comment by tobz** [comment]
This looks like a visual artifact of a datapoint being reported (on the far left), and then not being reported for a while, and then being reported with a really high value... so `monotonic_diff` indeed shows the correct delta between the two data points, but connects those datapoints with a line that makes it appear like it's been increasing over time.

Can you confirm whether or not there are individual data points reported along that line that occur between the start and end time of the series?

**Reply by jszwedko** [reply]
The Sum and Max being identical seems to support this theory.

**Reply by leongshengmin** [reply]
@tobz can confirm that there should be individual data points. As sampling other pods from the same cluster, we're noticing the same monotonic increase as well up till the time the pod was restarted. (this time graphed the data points in the screenshots)
<img width="1553" alt="Screenshot 2022-10-07 at 2 53 48 PM" src="https://user-images.githubusercontent.com/35290582/194486581-92d834ee-005f-41f1-bdf3-54fc54e63566.png">
<img width="1553" alt="Screenshot 2022-10-07 at 2 53 01 PM" src="https://user-images.githubusercontent.com/35290582/194486557-77abadae-6186-4a61-8858-b507669f969c.png">
